contracts/desiege/README.md

```diff
@@ -0,0 +1,185 @@
+[![Discord](https://badgen.net/badge/icon/discord?icon=discord&label)](https://discord.gg/uQnjZhZPfu)
+[![Twitter](https://badgen.net/badge/icon/twitter?icon=twitter&label)](https://twitter.com/LootRealms)
+
+![This is an image](./desiege.jpg)
+
+# 🛡️ Desiege
+
+### Desiege - A team-based coordination game built on Starknet.
+
+Desiege is a player-vs-player experience built on StarkNet. 
+
+**This folder contains Desiege Game contracts. If you're looking for another contract, please see the [directory of our Realms Smart Contracts](/).**
+
+### Game module overview
+
+| Module          | Function                             | Current Status |
+| --------------- | ------------------------------------ | -------------- |
+| [TowerDefence](./01_TowerDefence.cairo)        | The core Desiege contract | In review      |
+| [TowerDefenceStorage](./02_TowerDefenceStorage.cairo)       | Storage variables for Desiege                  | In review      |
+| [GridMovement](./03_GridMovement.cairo)       | Checks movement logic                 | In review      |
+| [Elements](./04_Elements.cairo)      | Users can mint Elements               | In review      |
+
+<hr>
+
+## Understanding the code
+<details><summary> 🤔 What is the game?</summary>
+
+<p>
+Players select a team (Light or Dark) and need to coordinate to ensure a victory for their team.
+
+Light players can cast a Shield spell and defend the city.
+
+Dark players can cast an Attack spell and destroy the city.
+
+</p>
+</details>
+<details><summary>🏗️ System architecture</summary>
+
```

Comment:
@micksabox could you inject the architecture about the contracts in here

Reply:
@ponderingdemocritus would you like for us to do this in the current PR or a separate PR once this lands?

Reply:
yeah let's land this PR, and task it out